include/seqan3/alignment/pairwise/alignment_result.hpp

```diff
@@ -210,36 +210,68 @@ class alignment_result
-                      "configuration.");
-        return data.back_coordinate;
+        static_assert(!std::is_same_v<size_t, std::nullopt_t *>,
+                      "Trying to access the end position of the first sequence, although it was not requested in the"
+                      " alignment configuration.");
+        return data.end_position_pair.first;
     }
 
-    /*!\brief Returns the front coordinate of the alignment.
+    /*!\brief Returns the end position of the second sequence of the alignment.
+     * \return A pair of positions in the respective sequences, where the calculated alignment ends (inclusive).
+     *
+     * \note This function is only available if the end position of the second sequence was requested via the
+     * alignment configuration (see seqan3::align_cfg::result).
+     */
+    constexpr size_t sequence2_end_position() const noexcept
+    {
+        static_assert(!std::is_same_v<size_t, std::nullopt_t *>,
+                      "Trying to access the end position of the second sequence, although it was not requested in the"
+                      " alignment configuration.");
+        return data.end_position_pair.second;
+    }
+
+    /*!\brief Returns the begin position of the first sequence of the alignment.
      * \return  A pair of positions in the respective sequences, where the calculated alignment starts.
      *
      * \details
      *
-     * Guaranteed to be smaller than or equal to `back_coordinate()`.
+     * Guaranteed to be smaller than or equal to `sequence1_end_position()`.
      *
-     * \note This function is only available if the front coordinate was requested via the alignment configuration
-     * (see seqan3::align_cfg::result).
+     * \note This function is only available if the begin position of the first sequence was requested via the
+     * alignment configuration (see seqan3::align_cfg::result).
+     */
+    constexpr size_t sequence1_begin_position() const noexcept
+    {
+        static_assert(!std::is_same_v<size_t, std::nullopt_t *>,
+                      "Trying to access the begin position of the first sequence, although it was not requested in the"
+                      " alignment configuration.");
+        return data.begin_position_pair.first;
+    }
+
+    /*!\brief Returns the begin position of the second sequence of the alignment.
+     * \return  A pair of positions in the respective sequences, where the calculated alignment starts.
+     *
+     * \details
+     *
+     * Guaranteed to be smaller than or equal to `sequence2_end_position()`.
+     *
+     * \note This function is only available if the begin position of the second sequence was requested via the
+     * alignment configuration (see seqan3::align_cfg::result).
      */
-    constexpr front_coord_t const & front_coordinate() const noexcept
+    constexpr size_t sequence2_begin_position() const noexcept
```

Comment:
Just a question, wouldn't it make more sense to define these begin functions before the end functions?

Reply:
Yes, I thought the same thing. However, we have type deductions with only back_coordinate but no front_coordinate. That is, we allow the result to include the back_coordinate without the front_coordinate. This leads me to leave the order as it is.

Reply:
Ok, makes sense. Thanks for your explanaition. :)